### PR TITLE
build: add instructions to rectify code that is not regenerated

### PR DESCRIPTION
**Pull request by otan**
Make it easier for the developer to know how to fix un-generated code by
specifying the required commands.

Release note: None

### CODE REVIEW

**Comment by otan** [comment]
(failure example: https://github.com/cockroachdb/cockroach/pull/56005)

File: build/teamcity-check.sh [hunk 1]
```diff
@@ -35,22 +37,23 @@ fi
 tc_start_block "Ensure generated code is up-to-date"
 # Buffer noisy output and only print it on failure.
 run build/builder.sh make generate &> artifacts/generate.log || (cat artifacts/generate.log && false)
-run build/builder.sh make buildshort &> artifacts/buildshort.log || (cat artifacts/buildshort.log && false)
 rm artifacts/generate.log
+check_clean "Run \`make generate\` to automatically fix these. For only BUILD.bazel issues, run \`bazel run //:gazelle\`."
```
Comment (irfansharif) [inline build/teamcity-check.sh:41]: I'd also be fine with suggesting `make bazel-generate` here.

**Review by irfansharif — APPROVED** [review]

File: build/teamcity-check.sh (hunk 1 above)
Comment (otan) [reply build/teamcity-check.sh:41]: have to teach people to stop using make somehow :P

**Comment by otan** [comment]
made `make bazel-generate` be a separate item, as otherwise it is only triggered on `make vendor_rebuild` and leads to a more confusing message.

**Comment by otan** [comment]
bors r=irfansharif

**Pull request merged**